tests/integration/pilot/common/routing.go

```diff
@@ -894,6 +896,126 @@ type vmCase struct {
 	host string
 }
 
+func flatten(clients ...[]echo.Instance) []echo.Instance {
+	instances := []echo.Instance{}
+	for _, c := range clients {
+		instances = append(instances, c...)
+	}
+	return instances
+}
+
+func DNSTestCases(apps *EchoDeployments) []TrafficTestCase {
+	makeSE := func(ips ...string) string {
+		return tmpl.MustEvaluate(`
+apiVersion: networking.istio.io/v1alpha3
+kind: ServiceEntry
+metadata:
+  name: dns
+spec:
+  hosts:
+  - "fake.service.local"
+  addresses:
+{{ range $ip := .IPs }}
+  - "{{$ip}}"
+{{ end }}
+  resolution: STATIC
+  endpoints: []
+  ports:
+  - number: 80
+    name: http
+    protocol: HTTP
+`, map[string]interface{}{"IPs": ips})
+	}
+	tcases := []TrafficTestCase{}
+	ipv4 := "1.2.3.4"
+	ipv6 := "1234:1234:1234::1234:1234:1234"
+	dummyLocalhostServer := "127.0.0.1"
+	cases := []struct {
+		name string
+		// TODO(https://github.com/istio/istio/issues/30282) support multiple vips
+		ips      string
+		protocol string
+		server   string
+		expected []string
```

Comment:
is expected ever different from ips?

Reply:
Not right now - I had some cases where there was some but they don't actually pass today